python/cudf/cudf/core/index.py

```diff
@@ -1462,14 +1404,32 @@ def notna(self):
     notnull = notna
 
     @_cudf_nvtx_annotate
-    def get_slice_bound(self, label, side, kind=None):
+    def get_slice_bound(self, label, side: str, kind=None):
         if kind is not None:
             warnings.warn(
                 "'kind' argument in get_slice_bound is deprecated and will be "
                 "removed in a future version.",
                 FutureWarning,
             )
-        return self._values.get_slice_bound(label, side, kind)
+        if side not in {"left", "right"}:
+            raise ValueError(f"Invalid side argument {side}")
+        if self.is_monotonic_increasing:
+            return self.searchsorted(label, side=side, ascending=True)
+        elif self.is_monotonic_decreasing:
+            return self.searchsorted(label, side=side, ascending=False)
```

Comment:
If `RangeIndex` offered a `searchsorted` method, then the `get_slice_bound` implementation could be promoted to the base class (a range index is always monotonic increasing or decreasing, so the `else` branch will never be hit).

Reply:
Were you thinking of adding a `searchsorted` method to `RangeIndex` here?

Reply:
I realised re-reading the code that I could do that, so I guess yes..

Reply:
Done.